pages/tutorials/graphql_sso.md.erb

```diff
@@ -0,0 +1,289 @@
+  ssoProviderUpdate(input:{
+    id:"<paste your provider id here>",
+    identityProvider:{
+      metadata:{
+        url:"https://dev-242943.oktapreview.com/app/exkfs9l426YMuKPnM0h7/sso/saml/metadata"
+      }
+    }
+  }) {
+    ssoProvider {
+      ...on SSOProviderSAML {
+        identityProvider {
+          issuer
+          certificate
+          ssoURL
+          metadata {
+            xml
+            url
+          }
+        }
+      }
+    }
+  }
+}
+
+```
+
+- can't do an update while the provider is enabled, need to disable it, make your changes, then re-test and re-enable it once it's all working
+
+
+## Advanced options
+
+- you don't need to set up a blank one if you already have your ID prov. information
+- set up your identity provider, set up your SSO in Buildkite with the details, go back to your identity provider and add the service url 
+
+- can have as many different providers enabled at the same time as you need, they will all be displayed to users before they go through the authorisation flow
+
+- can add a note to each provider login option in this list, to provide information to your users about each one 
+- can also set session duration
+- note and sessionduration can both be set on create and update
+- use the `note` field and `sessionDurationInHours` field respectively, both of which are top level attributes under the input object in the ssoprovidercreate and ssoproviderupdate mutations
+
+In the following example code, an existing identity provider is specified with the id field, and its note and session duration properties are updated:
+
+```graphql
+ssoProviderUpdate(input:{
+  id:"<paste your provider id here>",
+  note: ":ruby: Ruby team, please sign in here"
+  sessionDurationInHours: 72
+})
+```
+
+- full API documentation for the GraphQL SSO operations can be found in the GraphQL explorer
+- generate an API token from your personal settings page that has access to the graphql beta, and use that to log in to the graphql explorer
+- docs can be found in the top right corner, and are searchable
+
+## Other operations
+
+### Create a SAML provider with data
+
+- instead of using the create method alone, you can add your data to it immediately
+- use the create method
+- there are three data entry options: static data entry, xml blob, and xml url
+
+static data options:
+- identity provider (ssoURL and issuer and certificate)
+
+```graphql
+ssoProviderCreate(input:{
+  organizationId:"<paste your organization id here>",
+  type:SAML,
+  identityProvider:{
+    ssoURL:"https://okta.com/111111111",
+    issuer: "https://okta.com/aaaaaaaaa",
+    certificate:"---BEGIN CERT---zga9d6fa98f7sa987d9829h"
+  }
+})
+```
+
+xml blob options:
+- identity provider (metadata > xml)
+
+```graphql
+ssoProviderCreate(input:{
+  organizationId:"<paste your organization id here>",
+  type:SAML,
+  identityProvider:{
+    metadata:{
+      xml: "..."
+    }
+  }
+})
+```
+
+xml url options:
+- identity provider (metadata > url)
```

Comment:
I reckon this one would be good to roll into the main "creating a SSO provider" example, as it should be the main use case.

Reply:
I thought that people would still be creating a blank provider to start with? Don't people need to get the service provider information first, before they can complete their identity provider setup?